src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/PhysicalTableManager.java

```diff
@@ -0,0 +1,72 @@
+
+public class PhysicalTableManager {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PhysicalTableManager.class);
+
+    private final AmazonDynamoDbAdminUtils dynamoDbAdminUtils;
+    private final int pollIntervalSeconds;
+    private final Map<String, DynamoTableDescription> physicalTableDescriptions;
+    private final boolean canCreateTables;
+
+    public PhysicalTableManager(AmazonDynamoDB amazonDynamoDb, int pollIntervalSeconds, boolean canCreateTables,
+                                Collection<CreateTableRequest> tablesToCreateEagerly) {
+        this.dynamoDbAdminUtils = new AmazonDynamoDbAdminUtils(amazonDynamoDb);
+        this.pollIntervalSeconds = pollIntervalSeconds;
+        this.physicalTableDescriptions = new ConcurrentHashMap<>();
+        this.canCreateTables = canCreateTables;
+        if (canCreateTables && tablesToCreateEagerly != null && !tablesToCreateEagerly.isEmpty()) {
+            tablesToCreateEagerly.forEach(this::createTableIfNotExists);
+        }
+    }
+
+    /**
+     * Used by {@link TableMappingFactory} upon creation of a {@link TableMapping}, to ensure that the physical table
+     * that a virtual table is mapped to exists and to retrieve the table description from remote Dynamo.
+     * <p/>
+     * When we are allowed to create tables, then this will create the physical table if it doesn't already exist;
+     * otherwise, we assume that the physical table has already been created, and simply do a describe.
+     */
+    DynamoTableDescription ensurePhysicalTableExists(CreateTableRequest createTableRequest) {
+        return canCreateTables
+            ? createTableIfNotExists(createTableRequest)
+            : describeTable(createTableRequest.getTableName());
+    }
+
+    DynamoTableDescription describeTable(String tableName) {
+        return physicalTableDescriptions.computeIfAbsent(tableName, ignored ->
+            new DynamoTableDescriptionImpl(dynamoDbAdminUtils.describeTable(tableName)));
+    }
+
+    private DynamoTableDescription createTableIfNotExists(CreateTableRequest physicalTable) {
+        if (!canCreateTables) {
+            throw new UnsupportedOperationException("Cannot create physical tables in this mt-dynamo client");
+        }
+        final String tableName = physicalTable.getTableName();
+        return physicalTableDescriptions.computeIfAbsent(tableName, ignored ->
+            new DynamoTableDescriptionImpl(dynamoDbAdminUtils.describeTableIfExists(tableName)
+                .map(description -> {
+                    LOG.info(format("Using existing physical table %s", tableName));
+                    return description;
+                }).orElseGet(() -> {
+                    LOG.info(format("Creating physical table %s", physicalTable.getTableName()));
+                    dynamoDbAdminUtils.createTableIfNotExists(physicalTable, pollIntervalSeconds);
+                    return dynamoDbAdminUtils.describeTable(tableName);
```

Comment:
Nitpicking: this existed before this PR, but `AmazonDynamoDB#createTable` already returns `TableDescription`, we can use that to avoid extra remote call here.

Reply:
I started making changes for this but it turns out I have to change a lot more code in `AmazonDynamoDbAdminUtils`, so I'm going to leave it as it is for now... (`AmazonDynamoDbAdminUtils#createTableIfNotExists` waits until the table is active, and I think we might have to wait till the table is active before we do a describe on the table to make sure we get the `LatestStreamArn` of the table. The `TableDescription` that's in a `CreateTableResult` might not be sufficient.)